api/src/main/java/org/eclipse/microprofile/openapi/models/media/Schema.java

```diff
@@ -1192,4 +1201,929 @@ default Schema oneOf(List<Schema> oneOf) {
+        setBooleanSchema(booleanSchema);
+        return this;
+    }
+
+    /**
+     * Returns the examples property of this Schema instance.
+     *
+     * @return a copy List (potentially immutable) of example objects which this schema could describe
+     * @since 4.0
+     */
+    List<Object> getExamples();
+
+    /**
+     * Sets the examples property of this Schema instance.
+     *
+     * @param examples
+     *            a list of example objects which this schema could describe
+     * @since 4.0
+     */
+    void setExamples(List<Object> examples);
+
+    /**
+     * Sets the examples property of this Schema instance.
+     *
+     * @param examples
+     *            a list of example objects which this schema could describe
+     * @return current Schema instance
+     * @since 4.0
+     */
+    default Schema examples(List<Object> examples) {
+        setExamples(examples);
+        return this;
+    }
+
+    /**
+     * Adds an example to the examples list.
+     *
+     * @param example
+     *            the example to add to the examples list
+     * @return current Schema instance
+     * @since 4.0
+     */
+    Schema addExample(Object example);
+
+    /**
+     * Removes an example from the examples list.
+     *
+     * @param example
+     *            the example to remove from the examples list
+     * @since 4.0
+     */
+    void removeExample(Object example);
+
+    /**
+     * Gets a schema property by name.
+     * <p>
+     * Allows access to arbitrary properties in a schema object, allowing use of alternative schema dialects which use
+     * different property names (or the same property names with different data types).
+     * <p>
+     * When using the standard schema dialect, this method can be used to retrieve values set by other methods. E.g.
+     *
+     * <pre>
+     * {@code
+     * schema.setMinimum(new BigDecimal(3));
+     * BigDecimal minimum = (BigDecimal) schema.get("minimum"); // returns 3
+     * }
+     * </pre>
+     *
+     * @param propertyName
+     *            the property name
+     * @return the value of the named property, or {@code null} if a property with the given name is not set
+     */
+    Object get(String propertyName);
+
+    /**
+     * Sets a schema property.
+     * <p>
+     * Allows the modifications of arbitrary schema properties in a schema properties, allowing use of alternative
+     * schema dialects which use different property names (or the same property names with different data types).
+     * <p>
+     * Passing {@code null} as the {@code value} removes the property from the schema object.
+     * <p>
+     * {@code value} must be one of the following types, otherwise non-portable behavior results:
+     * <ul>
+     * <li>Any primitive type
+     * <li>Any primitive wrapper class
+     * <li>{@code null}
+     * <li>{@code String}
+     * <li>{@code BigDecimal}
+     * <li>{@code BigInteger}
+     * <li>Any type which {@link OASFactory} can create
```

Comment:
Noted that in the smallrye implementation, this would require quite a bit of refactoring since the class handling IO for each model type needs to have the IO class for every other model type passed in.

I still think it's a reasonable requirement, but would be open to changing it.

Reply:
Is this because of on-the-fly serialization in the smallrye PR?